smalltalksrc/Slang-Tests/SlangBasicTranslationTest.class.st

```diff
@@ -1155,6 +1155,163 @@ SlangBasicTranslationTest >> testGoto [
 	self assert: translation equals: 'goto lab'
 ]
 
+{ #category : #'tests-inline-builtins' }
+SlangBasicTranslationTest >> testInlineMethodSumArgumentsNoPrepare [
+	
+	| tMethod translation |
+	tMethod := (self getTMethodFrom: #methodSumParameters:to:with:).
+	generator doBasicInlining: true.
+	
+	translation := self translate: tMethod.
+	translation := translation trimBoth.
+	self
+		assert: translation
+		equals: '/* SlangBasicTranslationTestClass>>#methodSumParameters:to:with: */
+static sqInt
+methodSumParameterstowith(sqInt pFrom, sqInt pTo, sqInt anInteger)
+{
+	sqInt i;
+	sqInt pFrom1;
+	sqInt pTo1;
+
+	/* begin methodFrom:to: */
+	pFrom1 = pFrom + anInteger;
+	pTo1 = pTo + anInteger;
+	for (i = 0; i <= 5; i += 1) {
+		pTo1[i] = pFrom1[i];
```

Comment:
Does this compile? I think it does not, right? see https://godbolt.org/z/E4feMdv6E
=> it is testing the wrong behavior, and I don't like committing to wrong :).

I think we should remove this test.

We could fix it in a follow-up if needed

Reply:
Yes, I would remove them but I'd like to check later, because what I meant is something like this: https://godbolt.org/z/jn97EaP96

So right now the inlining logic does not "pass" the pragma types to the signature?